Tables support for IndexedVarArray

trulsf opened the pull request:
There will probably some alignment with JuMP if Tables support is added to JuMP (ref. https://github.com/jump-dev/JuMP.jl/issues/3096)

trulsf commented:
The latest changes depend on the availability of Tables support in JuMP (see https://github.com/jump-dev/JuMP.jl/pull/3104) and should await merging until the PR is merged.

hellemo commented on src/tables.jl:
```diff
@@ -1,107 +1,38 @@
-    end
-    if length(v) > 0 && !has_values(first(v).model)
-        error("No solution values available for variable")
+function _rows(x::Union{SparseArray,SparseVarArray,IndexedVarArray})
```
Could this be `AbstractSparseArray`? I guess we are about to delete SparseVarArray in another PR soon.

trulsf replied:
The implementation of _rows assumes a `data` property exists which is not available in general for AbstractSparseArray. Agree that SparseVarArray just can be removed here.

hellemo reviewed (approved):
Looks good!

hellemo commented:
> The latest changes depend on the availability of Tables support in JuMP (see [jump-dev/JuMP.jl#3104](https://github.com/jump-dev/JuMP.jl/pull/3104)) and should await merging until the PR is merged.

How about just adding rowtable without extending JuMP function while we wait to get all the other changes in and merged and have tests in place, and rework to extend JuMP methods after https://github.com/jump-dev/JuMP.jl/pull/3104 is merged, would that work?

trulsf commented:
I have moved rowtable out of JuMP.Containers and fixed tests to run with latest JuMP release. Maybe include this as part of 0.7 release and move back and fix tests when new JuMP version is available?

hellemo commented:
> I have moved rowtable out of JuMP.Containers and fixed tests to run with latest JuMP release. Maybe include this as part of 0.7 release and move back and fix tests when new JuMP version is available?

Thanks! Good idea, I originally planned to wait for the JuMP release, but I think as long as we don't export `rowtables` and make it clear that we don't support tables, we should be good. If we want to make it super clear we could rename it to`_rowtables`, but I don't think that's necessary.
